### PR TITLE
UX Elements updated

### PR DESCRIPTION
**Pull request by gloveboxes**
UX updated to Resources from Models. Note, file names etc not changed for now...

### CODE REVIEW

**Review by aaronpowell — APPROVED** [review]
Should we rename the razor files while we're at it?

File: src/AzureOpenAIProxy.Management/Components/Pages/ModelDelete.razor.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        {
+            NavigationManager.NavigateTo("/models");
+            return;
+        }
+
+        // check if the catalogId is in EventCatalogMap
+        var eventMap = await DbContext.Events.FirstOrDefaultAsync(x => x.Catalogs.Any(c => c.CatalogId == m.CatalogId));
+
+        // check if the catalogId is in Metric
+        var metric = await DbContext.Metrics.FirstOrDefaultAsync(x => x.CatalogId == m.CatalogId);
```
Comment (aaronpowell) [inline src/AzureOpenAIProxy.Management/Components/Pages/ModelDelete.razor.cs:62]: Shouldn't this be solved via cascading deletes, or did we remove relationships between the types?

Comment (aaronpowell) [reply src/AzureOpenAIProxy.Management/Components/Pages/ModelDelete.razor.cs:62]: Or is the goal to block deletes if there are metrics and/or events?

**Comment by gloveboxes** [comment]
> Should we rename the razor files while we're at it?

I'll raise an issue for it to be cleaned up.

Note, I've just added Delete support for Resources - It's not perfect, it checks if catalog_id is in use and silently fails (ie doesn't report the error and the resource is not deleted. This is more a function of my Blazor skills than desire.


image

**Review by aaronpowell — CHANGES_REQUESTED** [review]
I feel like this could be radically simplified by using [Message Box](https://mudblazor.com/components/messagebox#api) from MudBlazor, and have that all from the `ModelList` page.

File: src/AzureOpenAIProxy.Management/Components/Pages/ModelDelete.razor.cs (hunk 1 above)
Comment (gloveboxes) [reply src/AzureOpenAIProxy.Management/Components/Pages/ModelDelete.razor.cs:62]: > Shouldn't this be solved via cascading deletes, or did we remove relationships between the types?

I think we removed as we wanted historic record. But 100% sure that is the right decision...

Comment (gloveboxes) [reply src/AzureOpenAIProxy.Management/Components/Pages/ModelDelete.razor.cs:62]: > Or is the goal to block deletes if there are metrics and/or events?

That is what we decided. We want to keep historic data

**Comment by gloveboxes** [comment]
> I feel like this could be radically simplified by using [Message Box](https://mudblazor.com/components/messagebox#api) from MudBlazor, and have that all from the `ModelList` page.

I'm sure you are right - more a skills issue on my part so I hacked it 😱

**Review by aaronpowell — APPROVED** [review]

**Pull request merged**